File: LeetCode/219.Contains_Duplicate_II/219.Contains_Duplicate_II.py
#!/usr/bin/env python3


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        dic = {}  # num : index

        for i, n in enumerate(nums):
            if n in dic:
                if i - dic[n] <= k:
                    return True
            dic[n] = i

        return False

File: LeetCode/219.Contains_Duplicate_II/219.Contains_Duplicate_II.swift
class Solution {
    func containsNearbyDuplicate(_ nums: [Int], _ k: Int) -> Bool {
        var idxMap: [Int: Int] = [:]

        for (i, num) in nums.enumerated() {
            if let j = idxMap[num] {
                if i - j <= k {
                    return true
                }
            }
            idxMap[num] = i
        }

        return false
    }
}
